torchvision/transforms/functional_tensor.py

```diff
@@ -887,15 +887,33 @@ def _cast_squeeze_out(img: Tensor, need_cast: bool, need_squeeze: bool, out_dtyp
     return img
 
 
-def _apply_grid_transform(img: Tensor, grid: Tensor, mode: str) -> Tensor:
+def _apply_grid_transform(img: Tensor, grid: Tensor, mode: str, fill: Optional[List[float]]) -> Tensor:
 
     img, need_cast, need_squeeze, out_dtype = _cast_squeeze_in(img, [grid.dtype, ])
 
     if img.shape[0] > 1:
         # Apply same grid to a batch of images
         grid = grid.expand(img.shape[0], grid.shape[1], grid.shape[2], grid.shape[3])
+
+    # Append a dummy mask for customized fill colors, should be faster than grid_sample() twice
+    if fill is not None:
+        dummy = torch.ones((img.shape[0], 1, img.shape[2], img.shape[3]), dtype=img.dtype, device=img.device)
+        img = torch.cat((img, dummy), dim=1)
+
     img = grid_sample(img, grid, mode=mode, padding_mode="zeros", align_corners=False)
 
+    # Fill with required color
+    if fill is not None:
+        mask = img[:, -1:, :, :]  # N * 1 * H * W
+        img = img[:, :-1, :, :]  # N * C * H * W
+        mask = mask.expand_as(img)
+        fill_img = torch.tensor(fill, dtype=img.dtype, device=img.device).unsqueeze(0).unsqueeze(-1).unsqueeze(-1) \
```

Comment:
Maybe, we can replace this by `torch.tensor(fill, dtype=img.dtype, device=img.device).view(1, len(fill), 1, 1).expand_as(img)` to make the line shorter ?

Reply:
Yes, that can make it in one line. I should've thought of that...
Changed as requested in the latest commit.